org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/variables/IVariableFormatter.java

```diff
@@ -38,6 +38,15 @@
     void registerValueFormatter(IValueFormatter formatter, int priority);
 
     /**
+     * Get the matched value formatter for the specified type.
+     * @param type the type of the valuer
+     * @param options the options about the formatters
+     * @return the matched value formatter, if no formatter matches,
+     * <code>IllegalArgumentException</code> will be thrown
+     */
+    IValueFormatter getValueFormatter(Type type, Map<String, Object> options);
```

Comment:
valuetostring is designed for formatting an existing jdi value,  typetostring is designed for formatting the type of a jdi value, we can get a value formatter from a jdi value, and type formatter from a jdi type, on setVariable scenario, we need a value formatter from a jdi type to construct a jdi value, so the getValueFormatter(other another proper name) is required in VariableFormatter, we can remove the valuetostring, but we cannot avoid the usage of 
getValueFormatter